gen/service.go

```diff
@@ -243,14 +318,15 @@ func functionHelper(g Generator, s *compile.ServiceSpec, f *compile.FunctionSpec
 		TemplateFunc("wrapResponse", functionWrapResponse),
 		TemplateFunc("unwrapResponse", functionUnwrapResponse),
 		TemplateFunc("namePrefix", functionNamePrefix),
+		TemplateFunc("checkNoError", checkNoError),
 	)
 }
 
 // functionIsException generates an expression that provides the IsException
 // function for the given Thrift function.
 func functionIsException(g Generator, f *compile.FunctionSpec) (string, error) {
 	return g.TextTemplate(
-		`func(err error) bool {
+		`func(err <if (checkNoError)>interface{}<else>error<end>) bool {
```

Comment:
A thrift exception can be a struct or a pointer, so we should check that the exception is both a pointer _or_ non-pointer in the switch statement below.

In the generated code for `NoErrorService_GetValue_Helper` I'd expect something similar to:
```go
NoErrorService_GetValue_Helper.IsException = func(err interface{}) bool {
	switch err.(type) {
		case *NoErrorException, NoErrorException:
			return true
		default:
			return false
		}
	}
```

instead of just checking the pointer value.